### ISSUE TITLE
Network Feedback

### ISSUE THREAD

**Issue by neosr-project** [zengkun301/DCTLSA#1]
Hi, first of all thanks to everyone that worked on DCTLSA.
I want to give some feedback regarding this project: I've added it to [neosr](https://github.com/muslll/neosr) and trained both bicubic and realistic models with it. However, I made two small changes to it: replaced the attention function with [`scaled_dot_product_attention`](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) to improve training speeds, and added `dropout` after `out_B` (per research findings of ['Reflash Dropout'](https://github.com/XPixelGroup/RDSR)).
The comparisons bellow are from a model trained on downscaling algorithms only (to be specific nearest, bilinear, bicubic, lanczos and mitchell), using VGG Perceptual loss, [LDL](https://github.com/csjliang/LDL) and at end of training [FocalFrequency](https://github.com/EndlessSora/focal-frequency-loss) . The weights have been [released for public use](https://drive.google.com/drive/folders/1pGA25CgJeCpqZnFP_pwYY4669dGVkE-X) (CC0 license).

![dctlsa_cmp_4](https://github.com/zengkun301/DCTLSA/assets/132400428/5f36c5ce-7d68-47ac-b3d7-63ac57584fbf)

![dctlsa_cmp_2](https://github.com/zengkun301/DCTLSA/assets/132400428/c901764e-51f7-4ddf-9367-99a517dab8e9)

![dctlsa_cmp_3](https://github.com/zengkun301/DCTLSA/assets/132400428/e64d36db-6e40-4946-92dd-f381b5920366)

I also tested DCTLSA on complex realistic degradations (noise, compression, blur), and it performed very well:

![dctlsa_cmp_anime_2](https://github.com/zengkun301/DCTLSA/assets/132400428/34370685-8238-4564-b981-73c5f48f5292)

![dctlsa_cmp_anime_0](https://github.com/zengkun301/DCTLSA/assets/132400428/46091993-be32-460f-a68b-13465b46411e)

![dctlsa_cmp_anime_1](https://github.com/zengkun301/DCTLSA/assets/132400428/697e4112-9f8c-4053-959f-099c93f957da)

DCTLSA is a very training efficient network. Some areas for improvements I noticed:
- Despite being a lightweight network, inference speed is still slow, making it impractical for some user scenarios.
- There's a potential to improve the network restoration capability by increasing the receptive field, as recent research showed (see [OmniSR](https://github.com/Francis0625/Omni-SR) and [DAT](https://github.com/zhengchen1999/dat) and SRFormer [ConvFFN](https://github.com/HVision-NKU/SRFormer/blob/main/basicsr/archs/srformer_arch.py#L61)).
- Due to instabilities, the model can decrease the overall generated image white point, making it look darker. Adding some method to keep brightness consistency without using a new loss, such as what [SPAN](https://github.com/hongyuanyu/SPAN/blob/main/basicsr/archs/span_arch.py#L230) recently did, would be a nice step to improve it.

Thanks again to everyone that worked on this project.

**Comment by Shiqi72** [comment]
你好，我想请问一下Flops是如何计算的？我在复现代码时×2得到的flops与论文中不一致
_model = model.Model(args, checkpoint)
            input = torch.randn(1, 3, 170, 170).cuda()
            flops, params = profile(_model, inputs=(input, 0))
            print("flops", str(flops / 1e9))
            print("params", str(params / 1e6))

**Comment by zengkun301** [comment]
> 你好，我想请问一下Flops是如何计算的？我在复现代码时×2得到的flops与论文中不一致 _model = model.Model(args, checkpoint) input = torch.randn(1, 3, 170, 170).cuda() flops, params = profile(_model, inputs=(input, 0)) print("flops", str(flops / 1e9)) print("params", str(params / 1e6))

计算×2得到的flops，为了保证输出尺寸为1280x720，输入维度应为(1，3，640，360)。

**Comment by zengkun301** [comment]
> 你好，我想请问一下Flops是如何计算的？我在复现代码时×2得到的flops与论文中不一致 _model = model.Model(args, checkpoint) input = torch.randn(1, 3, 170, 170).cuda() flops, params = profile(_model, inputs=(input, 0)) print("flops", str(flops / 1e9)) print("params", str(params / 1e6))

你好。计算×2得到的flops，为了保证输出尺寸为1280x720，输入维度应为(1，3，640，360)。

**Comment by Shiqi72** [comment]
> > 你好，我想请问一下Flops是如何计算的？我在复现代码时×2得到的flops与论文中不一致 _model = model.Model(args, checkpoint) input = torch.randn(1, 3, 170, 170).cuda() flops, params = profile(_model, inputs=(input, 0)) print("flops", str(flops / 1e9)) print("params", str(params / 1e6))
> 
> 你好。计算×2得到的flops，为了保证输出尺寸为1280x720，输入维度应为(1，3，640，360)。

非常感谢您的解答，我在阅读论文时还有一个疑问，我想问一下SA可以提取全局特征，那在LFE阶段加入3×3深度卷积层扩大感受野具体有什么作用呢？

**Comment by zengkun301** [comment]
您好。LFE阶段加入3×3深度卷积层目的是扩大感受野的同时不引入更多的参数量（与一般卷积相比）。SA理论上能够提取全局特征，但实际应用中捕获全局信息能力有限。

**Comment by zengkun301** [comment]
> > > 你好，我想请问一下Flops是如何计算的？我在复现代码时×2得到的flops与论文中不一致 _model = model.Model(args, checkpoint) input = torch.randn(1, 3, 170, 170).cuda() flops, params = profile(_model, inputs=(input, 0)) print("flops", str(flops / 1e9)) print("params", str(params / 1e6))
> > 
> > 
> > 你好。计算×2得到的flops，为了保证输出尺寸为1280x720，输入维度应为(1，3，640，360)。
> 
> 非常感谢您的解答，我在阅读论文时还有一个疑问，我想问一下SA可以提取全局特征，那在LFE阶段加入3×3深度卷积层扩大感受野具体有什么作用呢？

您好。LFE阶段加入3×3深度卷积层目的是扩大感受野的同时不引入更多的参数量（与一般卷积相比）。SA理论上能够提取全局特征，但实际应用中捕获全局信息能力有限。
